src/lib/domain/PegInTransaction.ts

```diff
@@ -149,40 +146,114 @@ export default class PegInTransaction extends PegTransaction implements PegInTra
 	buildTransaction = (signature:string|undefined) => {
 		if (!this.ready) throw new Error('Not ready!');
 		if (signature) throw new Error('signature only for peg out!');
-		return { opReturn: this.buildOpReturn(), opDrop: this.buildOpDrop() };
+		return { opReturn: this.buildOpReturnTransaction(), opDrop: this.buildOpDropTransaction() };
+	}
+
+	private buildOpDropTransaction = () => {
+		if (!this.pegInData.stacksAddress) throw new Error('Stacks address required!');
+		const tx = new btc.Transaction({ allowUnknowOutput: true, lockTime: 0 });
+		this.addInputs(tx);
+		const asmScript = this.getOpDropP2shScript(this.pegInData.stacksAddress, this.pegInData.sbtcWalletAddress)
+		tx.addOutput({ script: asmScript, amount: BigInt(this.pegInData.amount) });
+		const changeAmount = Math.floor(this.maxCommit() - this.pegInData.amount - this.fee);
+		if (changeAmount > 0) tx.addOutputAddress(this.fromBtcAddress, BigInt(changeAmount), this.net);
+		return tx;
+	}
+
+	private buildOpReturnTransaction = () => {
+		if (!this.pegInData.stacksAddress) throw new Error('Stacks address required!');
+		const tx = new btc.Transaction({ allowUnknowOutput: true });
+		this.addInputs(tx);
+		// create a set of inputs corresponding to the utxo set
+		const data = this.buildData(this.pegInData.stacksAddress);
+		tx.addOutput({ script: btc.Script.encode(['RETURN', data]), amount: 0n });
+		if (isMini) {
+			const asmScript = btc.Script.encode([
+				'IF', 
+				btc.OutScript.encode(btc.Address(this.net).decode(this.pegInData.sbtcWalletAddress)),
+				'ELSE', 
+				144, 'CHECKSEQUENCEVERIFY', 'DROP',
+				btc.OutScript.encode(btc.Address(this.net).decode(this.fromBtcAddress)),
+				'CHECKSIG',
+				'ENDIF'
+			])
+			tx.addOutput({ script: asmScript, amount: BigInt(this.pegInData.amount) });
+		} else {
+			tx.addOutputAddress(this.pegInData.sbtcWalletAddress, BigInt(this.pegInData.amount), this.net);
```

Comment:
Wait, this function is named `buildOpReturnTransaction`, I guess it should use the old format and not the timelocked OP_DROP variant?